File: test/fixtures/syntax/v1/catalog.ts
import { BaseCatalog } from "./base-catalog.js";

export namespace Inventory {
  export type ItemKey = string;

  export enum ItemState {
    Ready = "ready",
    Retired = "retired",
  }

  export interface CatalogItem {
    readonly key: ItemKey;
    state: ItemState;
  }

  export class CatalogStore extends BaseCatalog {
    readonly items: CatalogItem[];

    constructor(items: CatalogItem[]) {
      super();
      this.items = items;
    }

    find(key: ItemKey): CatalogItem | undefined {
      return this.items.find((item) => item.key === key);
    }
  }

  export function createStore(items: CatalogItem[]): CatalogStore {
    return new CatalogStore(items);
  }
}

export default Inventory.CatalogStore;

File: test/fixtures/syntax/v1/Catalog.cs
using System;

namespace Fixture.Catalog;

public enum ItemState
{
    Ready,
    Retired
}

public interface ICatalog<T>
{
    T Find(string key);
}

public readonly record struct CatalogKey(string Value);

public sealed class Catalog<T> : ICatalog<T>
{
    public string Name { get; set; }

    public Catalog(string name)
    {
        Name = name;
    }

    public T Find(string key)
    {
        throw new NotImplementedException();
    }
}
